presto-json/src/test/java/com/facebook/presto/json/Person.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.json;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.json.JsonCodec;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class Person
+{
+    private String name;
+    private boolean rocks;
+    private Optional<String> lastName;
+
+    public static void validatePersonJsonCodec(JsonCodec<Person> jsonCodec)
+    {
+        // create object with null lastName
+        Person expected = new Person().setName("dain").setRocks(true);
+
+        String json = jsonCodec.toJson(expected);
+        assertFalse(json.contains("lastName"));
+        assertEquals(jsonCodec.fromJson(json), expected);
+
+        byte[] bytes = jsonCodec.toJsonBytes(expected);
+        assertEquals(jsonCodec.fromJson(bytes), expected);
+
+        // create object with missing lastName
+        expected.setLastName(Optional.empty());
+
+        json = jsonCodec.toJson(expected);
+        assertFalse(json.contains("lastName"));
+        assertEquals(jsonCodec.fromJson(json), expected);
+
+        bytes = jsonCodec.toJsonBytes(expected);
+        assertEquals(jsonCodec.fromJson(bytes), expected);
+
+        // create object with present lastName
+        expected.setLastName(Optional.of("Awesome"));
+
+        json = jsonCodec.toJson(expected);
+        assertTrue(json.contains("lastName"));
+        assertEquals(jsonCodec.fromJson(json), expected);
+
+        bytes = jsonCodec.toJsonBytes(expected);
+        assertEquals(jsonCodec.fromJson(bytes), expected);
+    }
+
+    public static void validatePersonListJsonCodec(JsonCodec<List<Person>> jsonCodec)
+    {
+        ImmutableList<Person> expected = ImmutableList.of(
+                new Person().setName("dain").setRocks(true),
+                new Person().setName("martin").setRocks(true),
```

Comment:
Would it make sense to have one entry with rocks == false, one more with empty last name and one more with present last name?